setup.py

```diff
@@ -14,167 +14,90 @@
 from setuptools.command.build_ext import build_ext
 
+from distutils.util import get_platform
 
-with open("pennylane_lightning/_version.py") as f:
-    version = f.readlines()[-1].split()[-1].strip("\"'")
-
-
-class get_pybind_include(object):
-    """Helper class to determine the pybind11 include path
-    The purpose of this class is to postpone importing pybind11
-    until it is actually installed, so that the ``get_include()``
-    method can be invoked."""
-
-    def __str__(self):
-        import pybind11
-
-        return pybind11.get_include()
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir = ""):
+        Extension.__init__(self, name, sources = [])
+        self.sourcedir = Path(sourcedir).absolute()
 
 
-def has_flag(compiler, flagname):
-    """Return a boolean indicating whether a flag name is supported on
-    the specified compiler.
+class CMakeBuild(build_ext):
     """
-    import tempfile
-    import os
-
-    with tempfile.NamedTemporaryFile("w", suffix=".cpp", delete=False) as f:
-        f.write("int main (int argc, char **argv) { return 0; }")
-        fname = f.name
-
-    try:
-        compiler.compile([fname], extra_postargs=[flagname])
-    except setuptools.distutils.errors.CompileError:
-        return False
-    finally:
-        try:
-            os.remove(fname)
-        except OSError:
-            pass
-
-    return True
-
-
-def cpp_flag(compiler):
-    """Return the -std=c++17 compiler flag.
+    This class is built upon https://github.com/diegoferigo/cmake-build-extension/blob/master/src/cmake_build_extension/build_extension.py and https://github.com/pybind/cmake_example/blob/master/setup.py
     """
-    flags = ["-std=c++17"]
-
-    for flag in flags:
-        if has_flag(compiler, flag):
-            return flag
 
-    raise RuntimeError("Unsupported compiler -- at least C++17 support is needed!")
+    user_options = build_ext.user_options + [
+        ("define=", "D", "Define variables for CMake")
+    ]
 
+    def initialize_options(self):
+        super().initialize_options()
+        self.define = None
 
-class BuildExt(build_ext):
-    """A custom build extension for adding compiler-specific options."""
+    def finalize_options(self):
+        # Parse the custom CMake options and store them in a new attribute
+        defines = [] if self.define is None else self.define.split(";")
+        self.cmake_defines = [f"-D{define}" for define in defines]
 
-    c_opts = {
-        "msvc": ["/EHsc", "/O2", "/W4", "/WX", "/std:c++17", "-D_USE_MATH_DEFINES"],
-        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra", "-Werror"],
-    }
+        super().finalize_options()
 
-    l_opts = {
-        "msvc": ["/WX"],
-        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra", "-Werror"],
-    }
-    if platform.system() == "Linux" and platform.machine() == "x86_64":
-        c_opts["unix"].append("-mavx")
```

Comment:
We should make sure the AVX build options get added somewhere here for x86_64 platforms.

Reply:
I agree. I will add some lines in `setup.py` and `CMakeFiles.txt` to deal with this.

Reply:
Fixed in 9322482ecd60f334a5f58aa8a386e291b058e9ba